tests/test_import_handler.py

```diff
@@ -0,0 +1,145 @@
+
+    @patch('DeviceManager.ImportHandler.db')
+    def test_restore_db_config(self, db_mock):
+        db_mock.session = AlchemyMagicMock()
+        self.assertIsNone(ImportHandler.restore_db_config())
+
+    @patch('DeviceManager.ImportHandler.db')
+    def test_save_templates(self, db_mock):
+        db_mock.session = AlchemyMagicMock()
+
+        json_payload = {'templates': [{'import_id': 1, 'attrs': [
+            {'label': 'temperature', 'type': 'dynamic', 'value_type': 'float'}]}], 'label': 'test_device', 'id': 1,
+            'created': '2019-08-29T18:18:07.801602+00:00'}
+
+        json_data = {"label": "test_device", "id": 1,
+                     "templates": [{'id': 1, 'label': 'test_template'}]}
+
+        result = ImportHandler.save_templates(json_data, json_payload)
+        self.assertIsNotNone(result)
+        self.assertTrue(result)
+
+        json_data = {"label": "test_device", "id": 1,
+                     "templates": [{'id': 2, 'label': 'test_template'}]}
+        result = ImportHandler.save_templates(json_data, json_payload)
+        self.assertIsNotNone(result)
+        self.assertFalse(result[0].attrs)
+
+    @patch('DeviceManager.ImportHandler.db')
+    def test_set_templates_on_device(self, db_mock):
+        db_mock.session = AlchemyMagicMock()
+
+        json_payload = {'templates': [{'import_id': 1, 'attrs': [
+            {'label': 'temperature', 'type': 'dynamic', 'value_type': 'float'}]}], 'label': 'test_device', 'id': 1,
+            'created': '2019-08-29T18:18:07.801602+00:00'}
+
+        saved_templates = [DeviceTemplate(label='test_template', attrs=[])]
+
+        self.assertIsNone(ImportHandler.set_templates_on_device(
+            Mock(), json_payload, saved_templates))
+
+    @patch('DeviceManager.ImportHandler.db')
+    def test_save_devices(self, db_mock):
+        db_mock.session = AlchemyMagicMock()
+
+        json_payload = {"devices": [{"id": "68fc", "label": "test_device_0"}, {
+            "id": "94dc", "label": "test_device_1"}]}
+        json_data = {"devices": [{"id": "68fc", "label": "test_device_0"}, {
+            "id": "94dc", "label": "test_device_1"}]}
+
+        saved_templates = [DeviceTemplate(label='test_template', attrs=[])]
+
+        result = ImportHandler.save_devices(
+            json_data, json_payload, saved_templates)
+        self.assertIsNotNone(result)
+        self.assertTrue(result)
+
+    def test_notifies_creation_to_kafka(self):
+        with patch('DeviceManager.ImportHandler.serialize_full_device') as mock_serialize_device_wrapper:
+            mock_serialize_device_wrapper.return_value = {'templates': [369], 'label': 'test_device', 'id': 1,
+                                                          'created': '2019-08-29T18:18:07.801602+00:00'}
+
+            with patch.object(KafkaInstanceHandler, "getInstance", return_value=MagicMock()):
+                ImportHandler().notifies_creation_to_kafka(
```

Comment:
Could you see if the expected occurred?

Reply:
I added a check that validates if the function passed the kafka flush create event in the next commit. Thanks!